pkg/kubectl/cmd/apply.go

```diff
@@ -94,14 +94,15 @@ func NewCmdApply(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	usage := "that contains the configuration to apply"
 	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
 	cmd.MarkFlagRequired("filename")
+	cmd.Flags().Bool("dry-run", false, "If true, only print validation for objects that would be created.")
```

Comment:
since we will add this flag in multiple commands, please create a helper method like `AddDryRunFlag` in `cmdutil`